test/utils/utils.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"golang.org/x/oauth2/google"
+	alpha "google.golang.org/api/compute/v0.alpha"
+	beta "google.golang.org/api/compute/v0.beta"
+	"google.golang.org/api/compute/v1"
+)
+
+var (
+	currentNodeCountRE = regexp.MustCompile("currentNodeCount: (.*)\n")
+
+	errNoMatchFound        = errors.New("no match found for the given regex")
```

Comment:
Do we really need static errs here? vs just do fmt.Errorf?
Who is going to consume these?

Reply:
They were used in test to make sure we are failing correctly. Since now we use list format directly, the helper function and its test are removed, and these errors are turned into inline fmt.Errorf.

Reply:
I think it's overkill. The test can just check that an error was detected.